lib/datasource/docker.js

```diff
@@ -28,6 +29,9 @@ async function getAuthHeaders(registry, repository) {
   try {
     const apiCheckUrl = `${registry}/v2/`;
     const apiCheckResponse = await got(apiCheckUrl, { throwHttpErrors: false });
+    if (apiCheckResponse.headers['www-authenticate'] === undefined) {
+      return {};
```

Comment:
Don't we still need `Accept: 'application/vnd.docker.distribution.manifest.v2+json'` ?

Reply:
I have not seen a clear statement on it in the documentation. But this should still be used, otherwise the API server will most likely fall back to v1 of the schema.

Reply:
Do you have an actual server/service to test this against? e.g. Redhat? If it works fine with the v2 header then I think we should include it.

Reply:
I have successfully tested this using the image `registry-access-redhat-remote.bahnhub.tech.rz.db.de/jboss-eap-7/eap70-openshift:latest`. They don't implement the token workflow of the docker registry. I have already included as a separate unit test.